packages/help-center/src/hooks/use-messaging-availability.ts

```diff
@@ -0,0 +1,47 @@
+import config from '@automattic/calypso-config';
+import { useQuery } from '@tanstack/react-query';
+import apiFetch from '@wordpress/api-fetch';
+import { addQueryArgs } from '@wordpress/url';
+import wpcomRequest, { canAccessWpcomApis } from 'wpcom-proxy-request';
+import type { MessagingAvailability } from '../types';
+
+interface APIFetchOptions {
+	global: boolean;
+	path: string;
+}
+
+async function requestMessagingAvailability() {
+	const currentEnvironment = config( 'env_id' );
+	const params = {
+		group: 'wpcom_messaging',
+		environment: currentEnvironment === 'development' ? 'development' : 'production',
+	};
+	const wpcomParams = new URLSearchParams( params );
+	return canAccessWpcomApis()
+		? ( ( await wpcomRequest( {
+				path: '/help/messaging/is-available',
+				query: wpcomParams.toString(),
+				apiNamespace: 'wpcom/v2',
+				apiVersion: '2',
+				method: 'GET',
+		  } ) ) as MessagingAvailability )
+		: ( ( await apiFetch( {
+				path: addQueryArgs( '/help-center/support-availability/messaging', params ),
+				method: 'GET',
+				global: true,
+		  } as APIFetchOptions ) ) as MessagingAvailability );
```

Comment:
```suggestion
	return canAccessWpcomApis()
		? await wpcomRequest< MessagingAvailability >( {
				path: '/help/messaging/is-available',
				query: wpcomParams.toString(),
				apiNamespace: 'wpcom/v2',
				apiVersion: '2',
				method: 'GET',
		  } )
		: await apiFetch< MessagingAvailability >( {
				path: addQueryArgs( '/help-center/support-availability/messaging', params ),
				method: 'GET',
				global: true,
		  } as APIFetchOptions );
```

The above suggestions in one edit.

Reply:
This was specifically referring to Omar's suggestions. Haha just a rewrite that was easier to read.

Reply:
Fixed as per https://github.com/Automattic/wp-calypso/pull/76777#discussion_r1196209571 🙇